crypto/ts/ts_rsp_sign.c

```diff
@@ -58,46 +59,27 @@ static ASN1_INTEGER *def_serial_cb(struct TS_resp_ctx *ctx, void *data)
     return NULL;
 }
 
-#if defined(OPENSSL_SYS_UNIX)
-
 static int def_time_cb(struct TS_resp_ctx *ctx, void *data,
                        long *sec, long *usec)
 {
+    OSSL_TIME t;
     struct timeval tv;
-    if (gettimeofday(&tv, NULL) != 0) {
-        ERR_raise(ERR_LIB_TS, TS_R_TIME_SYSCALL_ERROR);
-        TS_RESP_CTX_set_status_info(ctx, TS_STATUS_REJECTION,
-                                    "Time is not available.");
-        TS_RESP_CTX_add_failure_info(ctx, TS_INFO_TIME_NOT_AVAILABLE);
-        return 0;
-    }
-    *sec = tv.tv_sec;
-    *usec = tv.tv_usec;
 
-    return 1;
-}
-
-#else
-
-static int def_time_cb(struct TS_resp_ctx *ctx, void *data,
-                       long *sec, long *usec)
-{
-    time_t t;
-    if (time(&t) == (time_t)-1) {
+    t = ossl_time_now();
+    if (ossl_time_is_zero(t)) {
```

Comment:
Will `ossl_time_now()` ever really fail? There was a comment above that checking the return value of `time(NULL)`  for `-1` makes no sense.

Reply:
Yeah, saw that comment.  Not sure I agree: if a system call is documented as being able to return failure, it should be checked for IMO.  Well apart from perhaps `close(2)` where there is no recovery possible.